src/main.js

```diff
@@ -1,12 +1,21 @@
-import cli from 'cli';
+import { getConfig } from 'cli';
 import Linter from 'linter';
 import log from 'logger';
 
 import 'babel-polyfill';
 
-export function createInstance({config=cli.argv, runAsBinary=false} = {}) {
+
+export function isRunFromCLI(_module=module) {
+  return require.main === _module;
+}
+
+export function createInstance({
+  config=getConfig(isRunFromCLI()).argv, runAsBinary=false,
```

Comment:
what was your reasoning behind adding `isRunFromCLI()`? For added convenience?

In the case of `web-ext` it's not necessary because we will make the call like this:

````
const linter = createInstance({config: ourLocalConfig});
````

This would mean `getConfig()` never gets called and yargs options are never added. If someone were to not pass in their own config then I just don't see how this is useful. They would be stuck with a dummy config that does nothing.